browser/ui/webui/ai_chat/ai_chat_ui.cc

```diff
@@ -93,6 +94,16 @@ AIChatUI::AIChatUI(content::WebUI* web_ui)
 #endif
 
   untrusted_source->AddBoolean("isMobile", kIsMobile);
+  content::WebContents* chat_web_contents = GetChatWebContents();
+  size_t visible_conversation_history_size = 0;
+  if (chat_web_contents) {
+    ai_chat::AIChatTabHelper* active_chat_tab_helper =
+        ai_chat::AIChatTabHelper::FromWebContents(chat_web_contents);
+    visible_conversation_history_size =
+        active_chat_tab_helper->GetVisibleConversationHistory().size();
+  }
+  untrusted_source->AddBoolean("hasInitialHistory",
+                               visible_conversation_history_size != 0);
```

Comment:
I don't think this needs to be provided by the C++. The context can just check if the first call to get conversation history has any data.

Reply:
@petemill what particular call do you mean?

Reply:
I think @petemill means [here](https://github.com/brave/brave-core/blob/42164aef57a52f588bd52316c8136d21cc79615b/components/ai_chat/resources/page/state/data-context-provider.tsx#L133) where we call `getConversationHistory`

If its the first call, and we get nothing back, we don't have initial history.

Reply:
@fallaciousreasoning the `getConversationHistory` returns 0 always even if we have history.

Reply:
Surely there's some way that we load the active conversation history because we display it, right?